### PR TITLE
Use string interpolation when possible

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/NodaTime.TzdbCompiler/Tzdb/ParserHelper.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ internal static long ConvertSecondsWithFractionalToTicks(string text)
                                          CultureInfo.InvariantCulture);
             if (number < 0.0 || number >= 60.0)
             {
-                throw new FormatException("seconds out of valid range of [0, 60): " + number);
+                throw new FormatException($"seconds out of valid range of [0, 60]: {number}");
```
Comment (Youssef1313) [inline src/NodaTime.TzdbCompiler/Tzdb/ParserHelper.cs]: There was a small typo here.

**Comment by jskeet** [comment]
Please be aware that I'm not going to get a chance to review these multiple PRs probably for another week. I'll get to them eventually, but over the holidays my family takes priority.

File: src/NodaTime.TzdbCompiler/Tzdb/ParserHelper.cs (hunk 1 above)
Comment (malcolmr) [reply src/NodaTime.TzdbCompiler/Tzdb/ParserHelper.cs]: That's not a typo: `[a, b)` means a value _x_ such that _a_ ≤ _x_ < _b_. See e.g. https://en.wikipedia.org/wiki/Interval_(mathematics)#Terminology.

In this case, 0 is a valid number of seconds, but 60 isn't (but 59.9 would be).

Comment (Youssef1313) [reply src/NodaTime.TzdbCompiler/Tzdb/ParserHelper.cs]: @malcolmr. Thanks for correction. I'm going to revert my bad change.

File: src/NodaTime/TimeZones/IO/DateTimeZoneReader.cs [hunk 2]
```diff
@@ -152,7 +152,7 @@ public int ReadMilliseconds()
                             millis = (firstData << 24) + (ReadByte() << 16) + (ReadInt16() & 0xffff);
                             break;
                         default:
-                            throw new InvalidNodaDataException("Invalid flag in offset: " + flag.ToString("x2"));
+                            throw new InvalidNodaDataException($"Invalid flag in offset: {flag.ToString("x2")}");
```
Comment (jskeet) [inline src/NodaTime/TimeZones/IO/DateTimeZoneReader.cs]: This would be better as `$"Invalid flag in offset: {flag:x2}"`

**Pull request merged**